backend/remote/backend.go

```diff
@@ -149,8 +149,16 @@ func (b *Remote) configure(ctx context.Context) error {
 	b.hostname = d.Get("hostname").(string)
 	b.organization = d.Get("organization").(string)
 
+	// The backend schema for the workspaces attribute changed in terraform
+	// 0.12, and causes a panic when encountered by 0.11. While we don't support
+	// downgrading, we do want to protect users from that panic.
+	wsConfig := d.Get("workspaces").(*schema.Set).List()
+	if len(wsConfig) == 0 {
+		return fmt.Errorf("the cached backend configuration is not valid for this version of terraform")
```

Comment:
Agree that it's not too informative, but it gets the job done. Maybe making it a specific error is not a bad idea as this doesn't tell the user how to fix it (which is purging the cached config)?

I'm not sure how many people will switch to `0.12` and then back to `0.11`, so no idea if its worth it.

Reply:
I imagine this would only happen if someone was in the middle of testing or upgrading to 0.12 and accidentally switched back to the older binary - that's how I stumbled across it. 

I didn't even consider purging the cached config as an option; I guess if there's no remote state it's fine, but if there's any remote state terraform will fail anyway:
`Error refreshing state: Terraform 0.11.13 does not support state version 4, please update.`